### PR TITLE
When refreshing iptables, solve the risk of interruption between communication services

### LINKED ISSUE #71: When refreshing iptables, there is a risk of communication interruption between the two services that are communicating
**What happened**:
When refreshing iptables, there is a risk of communication interruption between the two services that are communicating.

```go
// Start iptables proxy
func (proxier *Proxier) Start() {
	go func() {
		ticker := time.NewTicker(5 * time.Minute)
		for {
			select {
			case <-ticker.C:
				proxier.FlushRules()     // interruption risk is here 
				proxier.EnsureRules()
			case <-beehiveContext.Done():
				proxier.FlushRules()
				return
			}
		}
	}()
}
```

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- EdgeMesh version:
- Kubernetes version (use `kubectl version`):
- KubeEdge version(e.g. `cloudcore --version` and `edgecore --version`):
- <details><summary>Cloud nodes Environment:</summary>

    - Hardware configuration (e.g. `lscpu`):
    - OS (e.g. `cat /etc/os-release`):
    - Kernel (e.g. `uname -a`):
    - Go version (e.g. `go version`):
    - Others:

  </details>
- <details><summary>Edge nodes Environment:</summary>

    - edgecore version (e.g. `edgecore --version`):
    - Hardware configuration (e.g. `lscpu`):
    - OS (e.g. `cat /etc/os-release`):
    - Kernel (e.g. `uname -a`):
    - Go version (e.g. `go version`):
    - Others:

  </details>

### PR DESCRIPTION
**Pull request by yz271544**
What type of PR is this?
/kind bug

What this PR does / why we need it:
Problems with accessing external services through edgemesh at the edge
Which issue(s) this PR fixes:

Fixes #71

Special notes for your reviewer:

1.When getting edgemesh ignores routing rules and proxy rules, it also gets the routing rules that have been previously invalidated. These invalid routes need to be cleared first when refreshing the routing settings. Re-Ensure routing requires insert mode: utiliptables.Prepend
2. this is remove go-funk library version.

Signed-off-by: 胡正阳 huzhengyang@gridsum.com

### CODE REVIEW

**Comment by yz271544** [comment]
I also installed this golangci-lint tool locally, but I really can't understand its error message, and I feel like I have entered the text prison;  :tired_face:

**Comment by Poorunga** [comment]
Please use `make lint` to check code

It looks like you are not using the correct golang import order:
```
...
agent/pkg/proxy/iptables.go:6: File is not `gci`-ed with -local github.com/kubeedge (gci)
	beehiveContext "github.com/kubeedge/beehive/pkg/core/context"
	"github.com/kubeedge/edgemesh/agent/pkg/proxy/protocol"
	"github.com/kubeedge/edgemesh/common/util"
agent/pkg/proxy/iptables.go:16: File is not `gci`-ed with -local github.com/kubeedge (gci)
	"strings"
	"sync"
	"time"
...
```
CI: https://github.com/kubeedge/edgemesh/pull/78/checks?check_run_id=3532685541

File: agent/pkg/proxy/iptables.go [hunk 1]
```diff
@@ -3,21 +3,19 @@ package proxy
...
 	utiliptables "k8s.io/kubernetes/pkg/util/iptables"
 	utilexec "k8s.io/utils/exec"
-
-	beehiveContext "github.com/kubeedge/beehive/pkg/core/context"
-	"github.com/kubeedge/edgemesh/agent/pkg/proxy/protocol"
-	"github.com/kubeedge/edgemesh/common/util"
+	"strings"
+	"sync"
+	"time"
 )
```
Comment (Poorunga) [inline agent/pkg/proxy/iptables.go:19]:
```suggestion
import (
	"context"
	"fmt"
	"strings"
	"sync"
	"time"

	"github.com/vishvananda/netlink"
	corev1 "k8s.io/api/core/v1"
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
	"k8s.io/client-go/kubernetes"
	"k8s.io/klog/v2"
	utiliptables "k8s.io/kubernetes/pkg/util/iptables"
	utilexec "k8s.io/utils/exec"

	beehiveContext "github.com/kubeedge/beehive/pkg/core/context"
	"github.com/kubeedge/edgemesh/agent/pkg/proxy/protocol"
	"github.com/kubeedge/edgemesh/common/util"
)
```

Comment (yz271544) [reply agent/pkg/proxy/iptables.go]: thank you!

File: common/util/util.go [hunk 2]
```diff
@@ -51,3 +52,104 @@ func GetPodsSelector(svc *v1.Service) labels.Selector {
 	}
 	return selector
 }
+
+// Contains returns true if an element is present in a iteratee.
+func Contains(in interface{}, elem interface{}) bool {
```
Comment (Poorunga) [inline common/util/util.go:57]: We should import package go-funk instead of copying the code.
(We can't copy the code, this is a non-compliant way.😢)

**Comment by yz271544** [comment]
The problem has been solved in 72, here is closed first.

**Pull request closed without merging**